server/neptune/gateway/event/source/credentials.go

```diff
@@ -0,0 +1,198 @@
+package source
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/bradleyfalzon/ghinstallation"
+	"github.com/google/go-github/v45/github"
+	"github.com/pkg/errors"
+)
+
+// GithubCredentials handles creating http.Clients that authenticate.
+type GithubCredentials interface {
```

Comment:
nit: should we keep this in the github package and remove github from the name of the structs and interfaces?

Reply:
I'm skipping duplicating this file in the neptune folder since I've changed structure to just pass installation token in. But the reason it wasn't in the github pkg was because this logic was for the gateway implementation (github is an internal pkg for the temporal workflows only)